### DISCUSSION TITLE
Allow aliases to be viewed in a table format

### DISCUSSION THREAD

**Discussion by wbevans** [simple-login/app#520] [Feature Requests]
This would allow you to see more details (such as alias notes and names) without expanding. Would also allow for sorting by any field.

**Comment by jumpingbrowntrout** [comment]
I support this feature request but have added another one which achieves the same and may be quicker to implement. Anyone interested in this request may like to support mine also #544

**Comment by WhamboMPS** [comment]
I am a huge fan of @wbevans' suggestion. I just migrated off of competitor [Spamex](https://www.spamex.com/), a service I have been using since 2004 (!) Spamex is rock solid but has had minimal improvements since it was created and I am delighted by the many new features of SimpleLogin. However, because I have been using this other service for so long I am bringing 400+ aliases that were being redirected through a custom subdomain I control with me in the migration. I find SimpleLogin's current card-only presentation of Aliases on the Dashboard to be the most problematical element of the service

Shown below is Spamex' UI equivalent for the SimpleLogin dashboard. I would love it if a user had a choice of Alias cards or Aliases shown in a table. Maintaining the sorting and filtering capabilities that SimpleLogin currently has should be a part of either UI presentation.

![Spamex List of Aliases](https://user-images.githubusercontent.com/88680678/129419793-58345655-6a7d-4614-93e1-c94b7949ad34.jpg)

**Comment by ghost** [comment]
Great idea! I originally tried to create good alias names for all of my aliases, but eventually gave up when I realized that you have to click "More" on every single card in order to be able to see them. I know you can still search, but it hides some of your best functionality. +1 to a table. 

You'd even get a lot more information on a single screen and it wouldn't look cluttered. Honestly, it would probably be much easier to implement than the current card system.

**Reply by nguyenkims** [reply]
@WhamboMPS  The card layout is used as it's also usable on mobile devices as opposed to a table layout which is hard to use on a small device. We can add more filtering options to make the UI more powerful. There's also this request https://github.com/simple-login/app/discussions/398 that would make organising aliases easier.

>  that you have to click "More" on every single card in order to be able to see them

@cppmaniac  There's a new option called "Always expand alias info" on the Settings page that allows all the info to be displayed without having to click on the "more" button.

**Reply by WhamboMPS** [reply]
Thanks @nguyenkims, this makes sense to me. I do believe that for the web UI you should offer the ability to toggle through both presentations. Just don't make table view available in mobile clients.

**Comment by WhamboMPS** [comment]
Consider the many-to-one implications (Associated Domain(s) to Alias) of issue #590 if this enhancement is added.

**Comment by lenaxia** [comment]
I opened https://github.com/simple-login/app/issues/635 and was pointed here, so reproducting my request here:

> Request:
> 
> Enable toggling between card view (current dashboard) and list view to enable high density review of all aliases. The current card view is nice, but is very cluttered making it hard to scan for specific things, e.g. reviewing what aliases you have.
> 
> Criteria:
> 
> * A toggle button in the top left or right that enables list or card view
> * List view is one line per alias
> * Columns are Enable, Alias, Description (non editable), Last activity, Send Email, "More V" button to show detail as is on current cards
> * Clicking the "More V" to expand the alias will make the Description editable
> * Columns should be sortable
> * Stretch goal: Each row should start with a check box for bulk actions

As mentioned, the related domain as a column is something I missed which is a good call out.

Having toggleable interfaces that are better for mobile vs desktop is an important quality of life improvement

**Reply by sabrant** [reply]
This is so needed!  I am new to the product and cannot get organized with the current unsortable card view process. I am sad this basic feature was requested so long ago and has not come to fruition.

**Comment by cautious-bread** [comment]
Adding my voice to this as I just added the same request over on the forum (I didn't know this discussion site was a thing).
https://forum.simplelogin.io/t/add-option-for-alternative-table-view-for-the-aliases-page-tab/537

The current Card View looks nice, but having the option to view all that info in a Table format would be so much nicer.
From a UX perspective this would allow better scanning of the information.

The default columns could be;
| Enable | Alias | Latest Sender and when | Description | Display Name |

With stats for forwarded, blocked, sent in the last 14 days.

Then hover icons for Pin, Transfer and Delete.
